### PR TITLE
Added friendslist option where users can add in new friends to their …

### PR DESCRIPTION
**Pull request by GSgiansen**
Friends list utiliy trial run, basic functionality of adding new friends done

### CODE REVIEW

File: lib/pages/friends.dart [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+            TextField(
+              controller: _friendIdController,
+              decoration: InputDecoration(
+                labelText: 'Friend ID',
+              ),
+            ),
+            SizedBox(height: 16.0),
```
Comment (bkjwjason) [inline lib/pages/friends.dart]: I think the sizedbox here causes pixel overflow, you can consider decreasing the `height` value? Or make it size according to screen height

File: lib/pages/friends.dart [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    final QuerySnapshot querySnapshot = await collection.get();
+    List<Friend> friendL = [];
+    for (var friend in friends) {
+
+      for (var doc in querySnapshot.docs) {
+        if (doc.id == friend) {
+          print("Friend ID: ${friend}");
+          print("Friend Data: ${doc.data()}");
+
+          friendL.add(Friend(
```
Comment (bkjwjason) [inline lib/pages/friends.dart]: I think you can consider to make this more dynamic

File: lib/pages/friends.dart [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  List<Friend> friendList = [];
+
+   // getfriends will quert firebase for friends list
+   //
+
+  //accepts a list of confirmed friend names, have to iterate across friends collection to get name
+  getFriendNames(List<dynamic> friends) async {
+    final CollectionReference collection = FirebaseFirestore.instance.collection('friends');
+    final QuerySnapshot querySnapshot = await collection.get();
+    List<Friend> friendL = [];
```
Comment (bkjwjason) [inline lib/pages/friends.dart:29]: Maybe for friendlist, you can have better naming conventions like `friendlst` or `flst`? `friendL` might be a bit confusing and not very clear when doing code tracing

**Review by bkjwjason — COMMENTED** [review]
I think some of the state names are a bit misleading and not very clear when doing code tracing. But well done!

File: lib/pages/friends.dart (hunk 3 above)
Comment (GSgiansen) [reply lib/pages/friends.dart:29]: i agree, will make it more readable

File: lib/pages/friends.dart (hunk 1 above)
Comment (GSgiansen) [reply lib/pages/friends.dart]: Yeap, i realised that when i changed phone screen during testing, will make it more dynamic

**Pull request merged**